src/LazyOperations/Complement.jl

```diff
@@ -112,3 +113,80 @@ We use the `isuniversal` method.
 function isempty(C::Complement)
     return isuniversal(C.X)
 end
+
+# -- Concrete complement implementations --
+
+"""
+    complement(X::LazySet)
+
+Return the complement of a set.
+
+### Input
+
+- `X` -- set
+
+### Output
+
+A `UnionSetArray` of half-spaces, i.e. the output is the the union of the linear
+constraints which are obtained by complementing each constraint of `X`.
+
+### Algorithm
+
+The principle used in this function is that if ``X`` and ``Y`` are any pair of sets,
+then ``(X ∩ Y)^C = X^C ∪ Y^C``. In particular, we can apply this rule for each constraint
+that defines a polyhedral set, hence the concrete complement can be represented as the set
+union of the complement of each constraint.
+"""
+function complement(X::LazySet)
+    return UnionSetArray(constraints_list(Complement(X)))
+end
+
+"""
+    complement(H::HalfSpace)
+
+Return the complement of a half-space.
+
+### Input
+
+- `H` -- halfspace
+
+### Output
+
+The halfspace that is complementary to `H`. If ``H: \\langle a, x \\rangle ≤ b``,
+then this function returns the halfspace ``H′: \\langle a, x \\rangle ≥ b``.
+(Note that complementarity is understood in a relaxed sense, since the intersection
+of ``H`` and ``H′`` is non-empty).
+"""
+function complement(H::HalfSpace)
+    return HalfSpace(-H.a, -H.b)
+end
+
+# --  Fallback implementation, requires constraints list of C.X --
+
+"""
+    constraints_list(C::Complement)
+
+Return the list of constraints of the complement of a set.
+
+### Input
+
+- `C` -- lazy set complement
+
+### Output
+
+A vector of linear constraints.
+
+### Notes
+
+The method requires that the list of constraints of the complemented set can be
+obtained. Then, each constraint is complemented and returned in the output array.
+The set union of such array corresponds to the concrete set complement.
+"""
+function constraints_list(C::Complement)
+    clist = constraints_list(C.X)
+    out = similar(clist)
+    for (i, ci) in enumerate(clist)
+        out[i] = complement(ci)
+    end
+    return out
+end
```

Comment:
Just a general comment: If we had a way to always receive a *copy* of the constraints, we could update them in-place here instead of creating a new copy in `complement`. But sets such as `HPolyhedron` give you their own constraints, so in-place modification would also modify the original set.

Reply:
i'm not sure if that is possible with `HalfSpace` because it's immutable and we need to change the `b`.

Reply:
Oh, true!